client/src/helpers/test/page-controller.spec.js

```diff
@@ -0,0 +1,129 @@
+import should from 'should'
+import pageController from '../page-controller'
+import { setSession } from './testHelper'
+import sKeys from '../session-keys'
+import { prepareAxiosResponse, createCompoundGetResponse  } from './axios-mock-helper'
+import mockData from './mockData.json'
+import StoreHelper from '../store-helper'
+import { commonBeforeEach } from './testHelper'
+
+jest.mock('axios')
+
+let router = {}
+router.push = jest.fn()
+
+const token = 'test'
+const refreshToken = 'testRefreshToken'
+
+const _beforeEach = async () => {
+  commonBeforeEach()
+  pageController.hasLoadedData = false
+  setSession()
+  const { demoData } = mockData
+  return prepareAxiosResponse('post', { 
+    visitId: mockData.visit._id, 
+    // visit: mockData.visit, 
+    token, 
+    demoData  
+  })
+    .then(() => {
+      return createCompoundGetResponse({demoData, visit: mockData.visit})
+        .then(() => {
+          return StoreHelper.fetchTokenData(token)
+        })
+    })
+}
+
+
+const _createRouteObject = (isDemo = false) => {
+  return {
+    query: { apiUrl: 'testApiUrl', token: refreshToken },
+    name: isDemo ? 'demo' : 'notDemo'
+  }
+}
+
+describe('page-controller tests', () => {
+  beforeEach((done) =>  _beforeEach().then(() => done()))
+
+  it('onPageChange [demo use-case]', async done => {
+    should.doesNotThrow(async () => {
+      const route = _createRouteObject(true)
+      await pageController.onPageChange(route)
+      should.exist(localStorage.getItem(sKeys.AUTH_TOKEN))
+      pageController.hasLoadedData.should.equal(true)
+      done()
+    })
+  })
+
+  it('onPageChange [standard use-case]', async done => {
+    should.doesNotThrow(async () => {
+      const route = _createRouteObject()
+      await pageController.onPageChange(route)
+      pageController.hasLoadedData.should.equal(true)
+      done()
+    })
+  })
+
+  it('_loadData (demo UC)', done => {
+    localStorage.setItem(sKeys.DEMO_TOKEN, 'demoToken')
+    const route = _createRouteObject(true)
+    should.doesNotThrow(async () => {
+      await pageController._loadData(route)
+      const demoData = localStorage.getItem('DemoData')
+      should.exist(demoData)
+      done()
+    })
+  })
+
+  it('_loadData (default UC)', done => {
+    const route = _createRouteObject(false)
+    localStorage.removeItem(sKeys.DEMO_TOKEN)
```

Comment:
Can you explain in a comment why there is a need to remove the demo token here?  It seems to be a dependency on the previous test. If this is true can we isolate the tests?  It'll male it feasible to skip a test if needed.

Reply:
Thank you for your comment! As it turns out, it wasn't really necessary, so I removed it... I think it does make it a bit more isolated...